custom-vu/src/main/kotlin/jces1209/vu/TakeScreenshotHook.kt

```diff
@@ -0,0 +1,46 @@
+package jces1209.vu
+
+import com.atlassian.performance.tools.jiraactions.api.ActionMetric
+import com.atlassian.performance.tools.jiraactions.api.WebJira
+import com.atlassian.performance.tools.jiraactions.api.measure.PostMetricHook
+import org.openqa.selenium.OutputType
+import org.openqa.selenium.TakesScreenshot
+import java.io.File
+import java.nio.file.Files
+import java.util.*
+
+/**
+ * Takes a post metric screenshot. This can help to identify how Jira looks after the action finished.
+ * It works/was tested only on [com.atlassian.performance.tools.virtualusers.api.browsers.GoogleChrome].
+ */
+class TakeScreenshotHook(
+    jira: WebJira
+) : PostMetricHook {
+    private val display = jira.driver as TakesScreenshot
+    private val directory = File(".")
+        .resolve("build")
+        .resolve("screenshots")
+
+    init {
+        Files.createDirectories(directory.toPath())
+    }
+
+    override fun run(actionMetricBuilder: ActionMetric.Builder) {
+        val actionMetric = actionMetricBuilder.build()
+        saveScreenshot(
```

Comment:
How much space did this take?
The idea was to take 1-3 screenshots per action type, not for every action metric. We produce 10-20k of metrics per cohort.

Reply:
I don't think it's useful for a standard regression test. You can take all the screenshots you need in 2 minutes test.

Reply:
And it works only with local VU (I can't imagine running 20k actions on a local VU).

Reply:
It can work on AWS too. We'd just have to put the screenshots under `test-results` dir.
It would let us avoid toggling this feature on and off all the time.
If it was constantly on, each report would include it locally. No need to go looking around for the page when you demoed it. It's gonna also help a ton when increasing action coverage.

Reply:
> We'd just have to put the screenshots under test-results dir.

I didn't know or I forgot about this contract.

Reply:
It is indeed tribal. My PR https://github.com/atlassian/virtual-users/pull/27/commits/271b3cf97ad9de7d338767033441255b7b440d3e makes it explicit.